src/API/Google/Middleware.php

```diff
@@ -552,4 +567,45 @@ public function is_subaccount(): bool {
 		// since transients don't support booleans, we save them as 0/1 and do the conversion here
 		return boolval( $is_subaccount );
 	}
+
+	/**
+	 * Performs a request to Google Shopping Data Integration (SDI) to get required information in order to form an auth URL.
+	 *
+	 * @return array An array with the JSON response from the WCS server.
+	 * @throws NotFoundExceptionInterface  When the container was not found.
+	 * @throws ContainerExceptionInterface When an error happens while retrieving the container.
+	 * @throws Exception When the response status is not successful.
+	 * @see google-sdi in google/services inside WCS
+	 */
+	public function get_sdi_auth_params() {
+		try {
+			/** @var Client $client */
+			$client   = $this->container->get( Client::class );
+			$result   = $client->get( $this->get_sdi_auth_endpoint() );
+			$response = json_decode( $result->getBody()->getContents(), true );
+
+			if ( 200 !== $result->getStatusCode() ) {
```

Comment:
Ah sorry! I realized that my simulation was not correct, because I changed this line from `if ( 200 !== $result->getStatusCode() ) {` to `if ( 200 === $result->getStatusCode() ) {` but then the Exception was thrown as 200 and that's why I was getting that result.